File: lab2_fabrica/fabrica/Foundation.cs
using System;
using System.Drawing;
using System.Runtime.Serialization;

namespace fabrica
{
    [DataContract]

    public class Foundation : DecorativeCosmetics
    {
        [DataMember]
        private  String coverage;
        [DataMember]
        private  String finish;

        public Foundation(String name, String brand, double price, String coverage, String finish,  String shade, String texture): base(name, brand, price, shade, texture)
        {
            this.coverage = coverage;
            this.finish = finish;
        }
        public String Coverage { 
            get{ return coverage; }
            set { coverage = value; }
        }
        public String Finish
        {
            get { return finish; }
            set { finish = value; }
        }

        public override string getinfo()
        {
            String info = "Name: " + name + "\r\n" +
                "Brand: " + brand + "\r\n" +
                "Price: " + price + "$" + "\r\n" +
                "Shade: " + shade + "\r\n" +
                "Texture: " + texture + "\r\n" +
                "Coverage: " + coverage + "\r\n" +
                "Finish: " + finish + "\r\n";
            return info;
        }

        public override Bitmap Img()
        {
            return (Bitmap)Bitmap.FromFile(".//img//foundation.png");
        }
    


    }
}

File: lab1 class/untitled/src/foundation.java
public class foundation extends DecorativeCosmetics{
    private final String coverage;
    private final String finish;

    public foundation(String name, String brand, double price, String coverage,  String finish, String scent, int shade, String texture) {
        super(name, brand, price,  shade, scent, texture);
        this.coverage=coverage;
        this.finish = finish;
    }
    public String getCoverage(){return coverage;}
    public String getFinish(){return finish;}
    @Override
    public void printinfo() {
        System.out.println(name);
        System.out.println("Brand: "+ brand);
        System.out.println("Price: "+ price+ "$");
        System.out.println("Coverage: " + coverage);
        System.out.println("Finish: " + finish);
        System.out.println("Scent: " + scent);
        System.out.println("Shade: " + shade);
        System.out.println("Texture: " + texture + "\n");
    }

}
